src/block/block_ext.c

```diff
@@ -999,23 +999,41 @@ __block_append(
      * object in the skiplist, check for a simple extension, and otherwise append a new structure.
      */
     if ((ext = el->last) != NULL && ext->off + ext->size == off)
+        /* ext is adjacent to off/size. */
         ext->size += size;
     else {
         ext = __block_off_srch_last(el->off, astack);
         if (ext != NULL && ext->off + ext->size == off)
+            /* ext is adjacent to off/size. */
             ext->size += size;
-        else {
+        else if (ext == NULL) {
             WT_RET(__wti_block_ext_alloc(session, &ext));
             ext->off = off;
             ext->size = size;
 
             for (i = 0; i < ext->depth; ++i)
                 *astack[i] = ext;
             ++el->entries;
-        }
 
-        /* Update the cached end-of-list */
-        el->last = ext;
+            /* Update the cached end-of-list */
+            el->last = ext;
+        } else if (ext->off >= (off + size)) {
+            /* ext is above off/size */
+            WT_RET(__wti_block_ext_alloc(session, &ext));
+            ext->off = off;
+            ext->size = size;
+
+            for (i = 0; i < ext->depth; ++i) {
+                ext->next[i] = *astack[i];
+                *astack[i] = ext;
+            }
+            ++el->entries;
+
+            /* Update the cached end-of-list */
+            el->last = ext;
+        } else
+            /* ext intersects or is below off/size */
```

Comment:
This is the scenario where the new off and size are not adjacent but are past the currently existing end of the extend list. Correct?

Reply:
This was supposed to be off/size either is before or intersects the last extent.  However, with the wrong if above it was actually off/size either intersects or is after the last extent.

However now that the preceeding else if is deleted this is everything other than an empty extent list or an extension of the last extent.